myriad-scheduler/src/main/java/org/apache/myriad/scheduler/NMExecutorCommandLineGenerator.java

```diff
@@ -0,0 +1,144 @@
+      public Protos.Environment.Variable apply(Map.Entry<String, String> x) {
+        return Protos.Environment.Variable.newBuilder().setName(x.getKey()).setValue(x.getValue()).build();
+      }
+    }));
+
+    StringBuilder yarnOpts = new StringBuilder();
+    String rmHostName = System.getProperty(KEY_YARN_RM_HOSTNAME);
+
+
+    if (StringUtils.isNotEmpty(rmHostName)) {
+      addJavaOpt(yarnOpts, KEY_YARN_RM_HOSTNAME, rmHostName);
+    }
+
+    if (yarnEnv.containsKey(KEY_YARN_HOME)) {
+      addJavaOpt(yarnOpts, KEY_YARN_HOME, yarnEnv.get("YARN_HOME"));
+    }
+
+    addJavaOpt(yarnOpts, KEY_YARN_NM_LCE_CGROUPS_HIERARCHY, "mesos/$TASK_DIR");
+    addJavaOpt(yarnOpts, KEY_NM_RESOURCE_CPU_VCORES, Integer.toString(profile.getCpus().intValue()));
+    addJavaOpt(yarnOpts, KEY_NM_RESOURCE_MEM_MB, Integer.toString(profile.getMemory().intValue()));
+    addJavaOpt(yarnOpts, KEY_NM_ADDRESS, ALL_LOCAL_IPV4ADDR + Long.toString(ports.get(0).getPort()));
+    addJavaOpt(yarnOpts, KEY_NM_LOCALIZER_ADDRESS, ALL_LOCAL_IPV4ADDR + Long.toString(ports.get(1).getPort()));
+    addJavaOpt(yarnOpts, KEY_NM_WEBAPP_ADDRESS, ALL_LOCAL_IPV4ADDR + Long.toString(ports.get(2).getPort()));
+    addJavaOpt(yarnOpts, KEY_NM_SHUFFLE_PORT, Long.toString(ports.get(3).getPort()));
+
+    if (myriadConfiguration.getYarnEnvironment().containsKey(ENV_YARN_NODEMANAGER_OPTS)) {
+      yarnOpts.append(" ").append(yarnEnv.get(ENV_YARN_NODEMANAGER_OPTS));
+    }
+    builder.addAllVariables(Collections.singleton(
+            Protos.Environment.Variable.newBuilder()
+                .setName(ENV_YARN_NODEMANAGER_OPTS)
+                .setValue(yarnOpts.toString()).build())
+    );
+    return builder.build();
+  }
+
+  protected void appendCgroupsCmds(StringBuilder cmdLine) {
+    //These can't be set in the environment as they require commands to be run on the host
+    if (myriadConfiguration.getFrameworkSuperUser().isPresent() && myriadConfiguration.isCgroupEnabled()) {
+      cmdLine.append(" export TASK_DIR=`basename $PWD`&&");
+      //The container executor script expects mount-path to exist and owned by the yarn user
+      //See: https://hadoop.apache.org/docs/stable/hadoop-yarn/hadoop-yarn-site/NodeManagerCgroups.html
+      //If YARN ever moves to cgroup/mem it will be necessary to add a mem version.
+      appendSudo(cmdLine);
+      cmdLine.append("chown " + myriadConfiguration.getFrameworkUser().get() + " ");
+      cmdLine.append(myriadConfiguration.getCGroupPath());
+      cmdLine.append("/cpu/mesos/$TASK_DIR &&");
+      cmdLine.append(String.format("export %s=%s -D%s=%s", ENV_YARN_NODEMANAGER_OPTS, ENV_YARN_NODEMANAGER_OPTS,
+          KEY_YARN_NM_LCE_CGROUPS_HIERARCHY, "mesos/$TASK_DIR"));
+    } else if (myriadConfiguration.isCgroupEnabled()) {
+      LOGGER.info("frameworkSuperUser not set ignoring cgroup configuration, this will likely can the nodemanager to crash");
```

Comment:
If it is highly likely/definite that the NodeManager will crash, I recommend setting a guard condition/exception handling within the configuration step.